### PR TITLE
draft Rules df

### PR DESCRIPTION
**Pull request by willLPD2I**
@tab1tha is this the kinda thing you mean?

### CODE REVIEW

File: cin_validator/cin_validator_class.py [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+
+    def create_rules_df(self):
+        # rules_broken = self.all_rules_issue_locs
+        rules_broken = (
+            self.all_rules_issue_locs["rule_code"].astype("str").unique().tolist()
+        )
+        print(rules_broken)
+        rule_messages = []
+        for rule in registry:
+            if str(rule.code) in rules_broken:
```
Comment (tab1tha) [inline cin_validator/cin_validator_class.py:103]: Hi Will, this works does work.

On second thought however, let us not loop through registry twice (we are doing that in the create_error_report_df method).
How about merging this feature with the create_error_report_df function such that when we identify the rules that fail, we immediately add the rule code-description pair to a  dictionary as you did with issue_dict?

What do you think? @willLPD2I

**Review by tab1tha — CHANGES_REQUESTED** [review]
Sidenote: I've merged in the parent branch so if you do `git merge main` on this branch, it'll filter out only the changes you've made.

File: cin_validator/cin_validator_class.py (hunk 1 above)
Comment (willLPD2I) [reply cin_validator/cin_validator_class.py:103]: @tab1tha Sounds way better! Cool!

**Review by tab1tha — APPROVED** [review]
This is good work Will.

**Comment by tab1tha** [comment]
Resolving the merge conflict introduced formatting differences so I'll run formatting now.

**Pull request merged**